tablo/models.py

```diff
@@ -189,74 +214,149 @@ def time_info(self):
-                params.append(start_time)
-                params.append(end_time)
-        elif self.start_time_field and not only_return_count and not additional_where_clause:
-            # If layer has a time component, default is to show first time step
-            where += " AND {layer_time_field} = (SELECT MIN({layer_time_field}) FROM {table})".format(
-                layer_time_field=self.start_time_field,
-                table=self.table
-            )
+            select_fields = 'COUNT(0)'
+        else:
+            # Delimit and alias fields for query in double quotes, but ignore '*'
+            select_fields = [f if '.' in f else 'source.{0}'.format(f) for f in return_fields]
+            select_fields = '", "'.join('"."'.join(f.split('.')) for f in select_fields).join('""')
+            select_fields = select_fields.replace('"*"', '*')
 
-        if object_ids:
-            where += ' AND {primary_key} IN ({object_id_list})'.format(
-                primary_key=PRIMARY_KEY_NAME,
-                object_id_list=','.join(['%s' for obj_id in object_ids]),
-            )
-            for obj_id in object_ids:
-                params.append(obj_id)
+            if return_geometry:
+                select_fields += ', ST_AsText(ST_Transform("source"."dbasin_geom", {0}))'.format(out_sr)
 
-        if extent:
-            where += ' AND ST_Intersects(dbasin_geom, ST_GeomFromText(%s, 3857)) '
-            params.append(extent.as_sql_poly())
+        join = '' if only_return_count else self._build_join_clause(return_fields, additional_where_clause)
+        where, query_params = self._build_where_clause(additional_where_clause, only_return_count, **kwargs)
 
         with get_cursor() as c:
             # where clause may contain references to table and fields
-            query_clause = 'SELECT {fields} FROM {table} WHERE ' + where
+            query_clause = 'SELECT {fields} FROM "{table}" AS "source" {join} {where}'
             query_clause = query_clause.format(
-                fields=','.join(return_fields),
-                table=self.table
+                fields=select_fields, table=self.table, join=join.strip(), where=where.strip()
             )
-            c.execute(query_clause, params)
+            c.execute(query_clause, query_params)
             response = dictfetchall(c)
 
         return response
 
+    def _build_join_clause(self, fields, where, **kwargs):
+        if where is None:
+            return ''
+
+        query_fields = set(fields)
+        query_fields = query_fields.union(self._parse_where_clause(where)[0])
+
+        join_fields = query_fields.intersection(self.related_fields.keys())
+        join_fields = set(f[:f.index('.')] for f in join_fields)  # Derive distinct table prefixes
+        join_clause = ''
+
+        for relation in self.featureservicelayerrelations_set.filter(related_title__in=join_fields):
+            join_clause += ' LEFT OUTER JOIN "{table}" AS "{related_title}"'.format(
```

Comment:
I can't tell if this where to reply to the query error you noted above, so I'll include two replies here:

Query Error: I removed a `where == '1=1'` check in `_parse_where_clause`, and now queries work if you provide `1=1`, `2=2`, `'A'='A'`, `1=1 AND 'A'='A'`, etc.

As for the `OUTER`, I have long been in the habit of including it in all my SQL queries for explicitness. I know it's not required after `LEFT`, `FULL` or `RIGHT`, but it indicates that it is not an `INNER` join... that said, if you feel strongly about it I'll remove it.